### PR TITLE
Objc sample

### PR DESCRIPTION
**Pull request by ramya25**
Moving swift sample to a directory
Adding MSAL objectiveC sample

### CODE REVIEW

File: iOS-swift/MSALiOS/Info.plist [hunk 1]
```diff
@@ -27,6 +27,11 @@
 	</array>
 	<key>CFBundleVersion</key>
 	<string>1</string>
+	<key>LSApplicationQueriesSchemes</key>
+	<array>
```
Comment (oldalton) [inline iOS-swift/MSALiOS/Info.plist]: this should be msauthv2 and msauthv3

File: iOS-ObjC/Podfile [hunk 2]
```diff
@@ -0,0 +1,11 @@
+# Uncomment the next line to define a global platform for your project
+# platform :ios, '9.0'
+
+target 'MSALiOSobjC' do
+  # Comment the next line if you don't want to use dynamic frameworks
+  use_frameworks!
+
+  # Pods for MSALiOSobjC
+  pod 'MSAL', '~> 0.5.0'
```
Comment (oldalton) [inline iOS-ObjC/Podfile]: please use latest MSAL instead (0.7.1)

File: iOS-ObjC/MSALiOSobjC/ViewController.m [hunk 3]
```diff
@@ -0,0 +1,278 @@
...
+
+/// 😃 A View Controller that will respond to the events of the Storyboard.
+
+@interface ViewController ()
+    
+    @end
+
+@implementation ViewController
+    
+    MSALPublicClientApplication *application;
```
Comment (oldalton) [inline iOS-ObjC/MSALiOSobjC/ViewController.m]: please declare this one in the interface instead

File: iOS-ObjC/MSALiOSobjC/ViewController.m [hunk 4]
```diff
@@ -0,0 +1,278 @@
...
+    
+    // Update the below to your client ID you received in the portal. The below is for running the demo only
+    NSString *clientId = @"66855f8a-60cd-445e-a9bb-8cd8eadbd3fa";
```
Comment (oldalton) [inline iOS-ObjC/MSALiOSobjC/ViewController.m]: please declare those as constants outside of implementation

File: iOS-ObjC/MSALiOSobjC/ViewController.m [hunk 5]
```diff
@@ -0,0 +1,278 @@
...
+         identifier within the directory itself (e.g. a domain associated to the
+         tenant, such as contoso.onmicrosoft.com, or the GUID representing the
+         TenantID property of the directory)
+         - error                The error that occurred creating the application object, if any, if you're
+         not interested in the specific error pass in nil.
+         */
+        
+        NSError *error = nil;
+        
+        NSURL *authorityURL = [NSURL URLWithString:(authorityString)];
```
Comment (oldalton) [inline iOS-ObjC/MSALiOSobjC/ViewController.m]: no need for curly braces around authorityString

File: iOS-ObjC/MSALiOSobjC/ViewController.m [hunk 6]
```diff
@@ -0,0 +1,278 @@
...
+        
+        MSALAuthority *authority = [MSALAuthority authorityWithURL:authorityURL error:nil];
```
Comment (oldalton) [inline iOS-ObjC/MSALiOSobjC/ViewController.m]: please use MSALAADAuthority instead

File: iOS-ObjC/MSALiOSobjC/ViewController.m [hunk 7]
```diff
@@ -0,0 +1,278 @@
...
+        
+        
+        MSALPublicClientApplicationConfig *pcaConfig = [[MSALPublicClientApplicationConfig alloc] initWithClientId:clientId
+                                                                                                       redirectUri:NULL
```
Comment (oldalton) [inline iOS-ObjC/MSALiOSobjC/ViewController.m]: please use nil instead of NULL

File: iOS-ObjC/MSALiOSobjC/ViewController.m [hunk 8]
```diff
@@ -0,0 +1,278 @@
...
+                                                                                                         authority:authority];
+        
+        application = [[MSALPublicClientApplication alloc] initWithConfiguration:pcaConfig error:&error];
+        
+        if (application == nil)
+        {
+            UIAlertView *alert = [[UIAlertView alloc] initWithTitle:@"Error initializing application"
```
Comment (oldalton) [inline iOS-ObjC/MSALiOSobjC/ViewController.m]: please use UIAlertController instead. Also why are we not logging it instead like in Swift sample?

File: iOS-ObjC/MSALiOSobjC/ViewController.m [hunk 9]
```diff
@@ -0,0 +1,278 @@
...
+            return;
+        }
+        
+        /**
+         Removes all tokens from the cache for this application for the provided account
+         
+         - account:    The account to remove from the cache
+         */
+        
+        [application removeAccount:account error:nil];
```
Comment (oldalton) [inline iOS-ObjC/MSALiOSobjC/ViewController.m]: track error and return value here

File: iOS-ObjC/MSALiOSobjC/ViewController.m [hunk 10]
```diff
@@ -0,0 +1,278 @@
...
+        _signout.enabled = false;
```
Comment (oldalton) [inline iOS-ObjC/MSALiOSobjC/ViewController.m]: please use NO instead. Also move all UI elements to helper methods like in latest Swift sample

File: iOS-ObjC/MSALiOSobjC/ViewController.m [hunk 11]
```diff
@@ -0,0 +1,278 @@
...
+         
+         - forScopes:           Permissions you want included in the access token received
+         in the result in the completionBlock. Not all scopes are
+         guaranteed to be included in the access token returned.
+         - account:             An account object that we retrieved from the application object before that the
+         authentication flow will be locked down to.
+         - completionBlock:     The completion block that will be called when the authentication
+         flow completes, or encounters an error.
+         */
+        NSArray *scopes = @[@"https://graph.microsoft.com/user.read"];
```
Comment (oldalton) [inline iOS-ObjC/MSALiOSobjC/ViewController.m]: scopes should be a constant

File: iOS-ObjC/MSALiOSobjC/ViewController.m [hunk 12]
```diff
@@ -0,0 +1,278 @@
...
+        MSALSilentTokenParameters *parameters = [[MSALSilentTokenParameters alloc] initWithScopes:scopes account:account];
+        
+        // interactionRequired means we need to ask the user to sign-in. This usually happens
+        // when the user's Refresh Token is expired or if the user has changed their password
+        // among other possible reasons.
+        
+        [application acquireTokenSilentWithParameters:parameters completionBlock:^(MSALResult *result, NSError *error)
+         {
+             if (error != nil){
+                 if (error.domain == MSALErrorDomain) {
```
Comment (oldalton) [inline iOS-ObjC/MSALiOSobjC/ViewController.m]: check both error and code in the same if

File: iOS-ObjC/MSALiOSobjC/ViewController.m [hunk 13]
```diff
@@ -0,0 +1,278 @@
...
+                         dispatch_async(dispatch_get_main_queue(), ^{
+                             [self acquireTokenInteractively];
+                             return;
+                         });
+                     }
+                 }
+             }
+             
+             [self updateResultView:result];
+             dispatch_async(dispatch_get_main_queue(), ^{
```
Comment (oldalton) [inline iOS-ObjC/MSALiOSobjC/ViewController.m]: please have helper methods for all UI updates like in Swift sample

File: iOS-ObjC/MSALiOSobjC/ViewController.m [hunk 14]
```diff
@@ -0,0 +1,278 @@
...
+                 _signout.enabled = true ;
+             });
+             [self contentWithToken:[result accessToken]];
+         }];
+        
+    }
+    
+    -(void)acquireTokenInteractively{
+        NSArray *scopes = @[@"https://graph.microsoft.com/user.read"];
```
Comment (oldalton) [inline iOS-ObjC/MSALiOSobjC/ViewController.m]: scopes should be a constant

File: iOS-ObjC/MSALiOSobjC/ViewController.m [hunk 15]
```diff
@@ -0,0 +1,278 @@
...
+        
+        
+        void (^completionBlock)(MSALResult *result, NSError *error) = ^(MSALResult *result, NSError *error) {
+            
+            dispatch_async(dispatch_get_main_queue(), ^{
+                
+                if (result)
+                {
+                    [self updateResultView:result];
+                    dispatch_async(dispatch_get_main_queue(), ^{
```
Comment (oldalton) [inline iOS-ObjC/MSALiOSobjC/ViewController.m]: multiple main queue dispatches here

File: iOS-ObjC/MSALiOSobjC/ViewController.m [hunk 16]
```diff
@@ -0,0 +1,278 @@
...
+     This will invoke the call to the Microsoft Graph API. It uses the
+     built in URLSession to create a connection.
+     */
+    
+    -(void) contentWithToken:(NSString *)token{
+        
+        // Specify the Graph API endpoint
+        NSURL *graphURL = [NSURL URLWithString:(graphURI)];
+        NSMutableURLRequest *graphRequest = [NSMutableURLRequest requestWithURL:graphURL];
+        NSString *authValue = [NSString stringWithFormat:@"Bearer %@", token];
```
Comment (oldalton) [inline iOS-ObjC/MSALiOSobjC/ViewController.m]: authvalue variable unused here

File: iOS-ObjC/MSALiOSobjC/ViewController.m [hunk 17]
```diff
@@ -0,0 +1,278 @@
...
+        // Set the Authorization header for the request. We use Bearer tokens, so we specify Bearer + the token we got from the result
+        [graphRequest addValue:[NSString stringWithFormat:@"Bearer %@", token] forHTTPHeaderField:@"Authorization"];
+        
+        
+        NSURLSession *session = [NSURLSession sharedSession];
+        NSURLSessionDataTask *dataTask = [session dataTaskWithRequest:graphRequest
+                                                    completionHandler:^(NSData *data, NSURLResponse *response, NSError *error) {
+                                                        NSLog(@"%@",data);
+                                                        if (error)
```
Comment (oldalton) [inline iOS-ObjC/MSALiOSobjC/ViewController.m]: code tabbing seems a bit off here

File: iOS-ObjC/MSALiOSobjC/ViewController.m [hunk 18]
```diff
@@ -0,0 +1,278 @@
...
+                                                        [self updateResultViewError:error];
+                                                        else {
+                                                            NSDictionary *json  = [NSJSONSerialization JSONObjectWithData:data options:0 error:nil];
+                                                            NSLog(@"%@",json);
+                                                            dispatch_async(dispatch_get_main_queue(), ^{
+                                                                [self.resultTextView setText:[json description]];
```
Comment (oldalton) [inline iOS-ObjC/MSALiOSobjC/ViewController.m]: update all UI elements in a helper method

File: iOS-swift/MSALiOS/Info.plist (hunk 1 above)
Comment (ramya25) [reply iOS-swift/MSALiOS/Info.plist]: updated

File: iOS-ObjC/Podfile (hunk 2 above)
Comment (ramya25) [reply iOS-ObjC/Podfile]: updated to 0.7.1

File: iOS-ObjC/MSALiOSobjC/ViewController.m (hunk 3 above)
Comment (ramya25) [reply iOS-ObjC/MSALiOSobjC/ViewController.m]: declared in interface

File: iOS-ObjC/MSALiOSobjC/ViewController.m (hunk 4 above)
Comment (ramya25) [reply iOS-ObjC/MSALiOSobjC/ViewController.m]: declared as constants outside of implementation

File: iOS-ObjC/MSALiOSobjC/ViewController.m (hunk 5 above)
Comment (ramya25) [reply iOS-ObjC/MSALiOSobjC/ViewController.m]: removed curly braces there

File: iOS-ObjC/MSALiOSobjC/ViewController.m (hunk 14 above)
Comment (ramya25) [reply iOS-ObjC/MSALiOSobjC/ViewController.m]: changed

File: iOS-ObjC/MSALiOSobjC/ViewController.m (hunk 15 above)
Comment (ramya25) [reply iOS-ObjC/MSALiOSobjC/ViewController.m]: removed

File: iOS-ObjC/MSALiOSobjC/ViewController.m (hunk 16 above)
Comment (ramya25) [reply iOS-ObjC/MSALiOSobjC/ViewController.m]: removed

File: iOS-ObjC/MSALiOSobjC/ViewController.m (hunk 18 above)
Comment (ramya25) [reply iOS-ObjC/MSALiOSobjC/ViewController.m]: updated

File: iOS-ObjC/MSALiOSobjC/ViewController.m (hunk 6 above)
Comment (ramya25) [reply iOS-ObjC/MSALiOSobjC/ViewController.m]: updated

File: iOS-ObjC/MSALiOSobjC/ViewController.m (hunk 7 above)
Comment (ramya25) [reply iOS-ObjC/MSALiOSobjC/ViewController.m]: updated

File: iOS-ObjC/MSALiOSobjC/ViewController.m (hunk 8 above)
Comment (ramya25) [reply iOS-ObjC/MSALiOSobjC/ViewController.m]: removed alert and logged it like in Swift sample

File: iOS-ObjC/MSALiOSobjC/ViewController.m (hunk 10 above)
Comment (ramya25) [reply iOS-ObjC/MSALiOSobjC/ViewController.m]: updated

File: iOS-ObjC/MSALiOSobjC/ViewController.m (hunk 11 above)
Comment (ramya25) [reply iOS-ObjC/MSALiOSobjC/ViewController.m]: changed as constant

File: iOS-ObjC/MSALiOSobjC/ViewController.m (hunk 12 above)
Comment (ramya25) [reply iOS-ObjC/MSALiOSobjC/ViewController.m]: checked in same if

File: iOS-ObjC/MSALiOSobjC/ViewController.m (hunk 17 above)
Comment (ramya25) [reply iOS-ObjC/MSALiOSobjC/ViewController.m]: Updated

File: iOS-ObjC/MSALiOSobjC/ViewController.m (hunk 13 above)
Comment (ramya25) [reply iOS-ObjC/MSALiOSobjC/ViewController.m]: updated

**Pull request closed without merging**